File: dev/osm2gmns_dev.cpp
//
// Created by Jiawei Lu on 2/17/23.
//

#include <absl/base/log_severity.h>
#include <absl/log/globals.h>
#include <absl/log/initialize.h>

#include <exception>
#include <filesystem>
#include <iostream>
#include <string>

#include "config.h"
#include "functions.h"
#include "io.h"
#include "networks.h"
#include "osmconfig.h"

int main(int /*argc*/, char* /*argv*/[]) {
  try {
    absl::InitializeLog();
    absl::SetStderrThreshold(absl::LogSeverityAtLeast::kInfo);

    // const auto map_folder = std::filesystem::path("maps/Texas");
    // const auto map_folder = std::filesystem::path("maps/Dallas");
    // const auto map_folder = std::filesystem::path("maps/Debug/Wickenburg");
    const auto map_folder = std::filesystem::path("maps/Debug/AZ");

    // const std::string map_filename = "map.osm.pbf";
    // const std::string map_filename = "northwest-latest.osm.pbf";
    // const std::string map_filename = "map.osm";
    const std::string map_filename = "arizona.pbf";

    const auto* osm_parsing_config =
        new OsmParsingConfig{{"highway", "ref"}, {"oneway", "junction", "lanes"}, {"building", "type"}};

    Network* network = getNetFromFile(map_folder / map_filename,
                                      {ModeType::AUTO, ModeType::BIKE, ModeType::RAILWAY, ModeType::AEROWAY},
                                      {HighWayLinkType::MOTORWAY, HighWayLinkType::TRUNK, HighWayLinkType::PRIMARY,
                                       HighWayLinkType::SECONDARY, HighWayLinkType::TERTIARY},
                                      {HighWayLinkType::RESIDENTIAL}, true, 1.0, osm_parsing_config, true);

    // consolidateComplexIntersections(network, false, map_folder / "intersection.csv");

    // generateNodeActivityInfo(network);

    // fillLinkAttributesWithDefaultValues(network, true, {}, false, {{HighWayLinkType::MOTORWAY, 1}}, true,
    //                                     {{HighWayLinkType::MOTORWAY, 1}});

    outputNetToCSV(network, map_folder);

    delete network;
  } catch (const std::exception& e) {
    std::cerr << e.what() << '\n';
    return 1;
  }
  return 0;
}

File: dev/osm2gmns_dev.py
import os
import sys
parent_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(parent_dir)

import osm2gmns as og

map_folder = r'dev/maps/yuba'

map_name = 'map.osm.pbf'

net = og.getNetFromFile(os.path.join(map_folder, map_name),
                        link_types=['motorway', 'trunk', 'primary', 'secondary'],
                        connector_link_types=['tertiary'], POI=True)

og.consolidateComplexIntersections(net, auto_identify=True)

og.generateNodeActivityInfo(net)

og.fillLinkAttributesWithDefaultValues(net, default_lanes=False, default_lanes_dict={},
                                       default_speed=True, default_speed_dict={},
                                       default_capacity=True, default_capacity_dict={'primary':1289})

og.outputNetToCSV(net, map_folder)

